### PR TITLE
fix: regression with tuple coords entries as xarray's (dim_name, values)

### PR DESCRIPTION
**Pull request by FBumann**
Mandatory to pull this in before next release.
Fixes a bug which was introduced by the alignment refactor. tuple coords were treated differently. Its now consistent with v0.7.0 and xarray.
And additionally improving and reorganizing test coverage

 > [!NOTE]
 > The following content was generated by AI.

 <details>
 <summary>Details</summary>

## What

`_coords_to_dict` treated a tuple coords entry as a bare value sequence (like a list), which broke the standard xarray form `coords=[("origin", origins)]`: a scalar bound against such coords raised an inscrutable numpy *"setting an array element with a sequence … inhomogeneous shape"* error.

```python
m.add_variables(lower=0, coords=[("origin", origins), ("dest", dests)])  # crashed
```

v0.7.0 delegated coords to xarray and supported this form; the alignment refactor in #742 (unreleased) regressed it and pinned the value-sequence reading in tests.

## Fix

Restore xarray's convention in `_coords_to_dict`:

- A **tuple** entry is `(dims, data[, attrs])` — its first element names the dimension.
- A **bare value sequence** uses a `list` (or `range` / `ndarray`).
- Length-1 / scalar-value tuples raise a clear `(dim_name, values)` error, mirroring xarray's own rejection.

Unnamed `pd.Index` entries are now named on a copy so an entry's `.name` matches its dim consistently across forms (no caller mutation).

## Tests

- Dropped the rows that pinned the wrong value-sequence reading; added `(name, values)` coverage (parse-level + end-to-end), the multi-tuple scalar-bound regression, and tuple edge cases.
- Restructured `test_alignment.py`'s coords tests into form-parameterized tables and split the end-to-end `add_variables` checks into their own class.

Full test suite passes (3787 passed, 29 skipped).

## Docs

Corrected the unreleased release note to state the xarray `(dim_name, values)` convention (the prior "bare-tuple entries behave like lists" wording described the regressed behaviour).

 </details>

### CODE REVIEW

File: test/test_alignment.py [hunk 1]
```diff
@@ -445,119 +441,172 @@ def test_explicit_dims_win_over_inference(self) -> None:
...
+            ([pd.Index([0, 1, 2], name="x")], None, ("x",)),
+            ([pd.Index([0, 1, 2])], ["x"], ("x",)),
+            ([[0, 1, 2]], ["x"], ("x",)),
+            ([range(3)], ["x"], ("x",)),
+            ([np.array([0, 1, 2])], ["x"], ("x",)),
+            ([[0, 1, 2]], None, ("dim_0",)),
+            ([range(3)], None, ("dim_0",)),
+            ([np.array([0, 1, 2])], None, ("dim_0",)),
+            ([pd.Index([0, 1, 2])], None, ("dim_0",)),
+            ([("origin", ["a", "b"]), ("dest", ["x", "y"])], None, ("origin", "dest")),
```
Comment (FabianHofmann) [inline test/test_alignment.py:588]: crazy how many combinations this supports :D

**Review by FabianHofmann — APPROVED** [review]
thanks for the follow-up and the tidying of the tests

File: test/test_alignment.py (hunk 1 above)
Comment (FBumann) [reply test/test_alignment.py:588]: Yes... 
But it always has. Its now just really obvious!
